subprojects/core/src/main/java/org/gradle/internal/service/scopes/GlobalScopeServices.java

```diff
@@ -287,6 +294,29 @@ ObjectFactory createObjectFactory(
             domainObjectCollectionFactory);
     }
 
+    @Provides
+    ExecFactory createExecFactory(
+        FileResolver fileResolver,
+        FileCollectionFactory fileCollectionFactory,
+        Instantiator instantiator,
+        ObjectFactory objectFactory,
+        ExecutorFactory executorFactory,
+        TemporaryFileProvider temporaryFileProvider,
+        BuildCancellationToken buildCancellationToken,
+        ClientExecHandleFactory execHandleFactory
+    ) {
+        return DefaultExecActionFactory.of(
+            fileResolver,
+            fileCollectionFactory,
+            instantiator,
+            executorFactory,
+            temporaryFileProvider,
+            buildCancellationToken,
+            objectFactory,
+            execHandleFactory
+        );
+    }
```

Comment:
❓ Why do we need a global `ExecFactory` now where we did not need one before?

Reply:
Before it was defined in `BasicGlobalScopeServices`, but we removed it from there